apps/asap-server/test/controllers/teams.test.ts

```diff
@@ -303,51 +303,90 @@ describe('Team controller', () => {
 
-      nock(config.baseUrl)
-        .post(`/api/content/${config.appName}/graphql`, {
-          query: buildGraphQLQueryFetchTeam(teamId),
-        })
-        .reply(200, graphQlTeamResponse)
-        .get(`/api/content/${config.appName}/users`)
-        .query({
-          $filter: `data/teams/iv/id eq '${teamId}'`,
-        })
-        .reply(200, userResponse);
+      test('Should parse the team user correctly when the avatar is null', async () => {
+        const teamId = 'team-id-1';
+
+        user.data.avatar.iv = null as any;
+
+        const userResponse = {
+          total: 1,
+          items: [user],
+        };
+
+        nock(config.baseUrl)
+          .post(`/api/content/${config.appName}/graphql`, {
+            query: buildGraphQLQueryFetchTeam(teamId),
+          })
+          .reply(200, graphQlTeamResponse)
+          .get(`/api/content/${config.appName}/users`)
+          .query({
+            $filter: `data/teams/iv/id eq '${teamId}'`,
+          })
+          .reply(200, userResponse);
+
+        const result = await teams.fetchById(teamId, mockUser);
+
+        const expectedResponse = {
+          ...fetchTeamByIdExpectation,
+          members: [
+            {
+              ...fetchTeamByIdExpectation.members[0],
+              avatarUrl: undefined,
+            },
+          ],
+        };
 
-      const result = await teams.fetchById(teamId, mockUser);
+        expect(result).toEqual(expectedResponse);
+      });
 
-      const expectedResponse = {
-        ...fetchTeamByIdExpectation,
-        members: [
-          {
-            ...fetchTeamByIdExpectation.members[0],
-            avatarUrl: undefined,
-          },
-        ],
-      };
+      test('Should parse the team user correctly when the avatar is undefined', async () => {
+        const teamId = 'team-id-1';
+
+        delete (user.data as any).avatar;
+
+        const userResponse = {
+          total: 1,
+          items: [user],
+        };
+
+        nock(config.baseUrl)
+          .post(`/api/content/${config.appName}/graphql`, {
+            query: buildGraphQLQueryFetchTeam(teamId),
+          })
+          .reply(200, graphQlTeamResponse)
+          .get(`/api/content/${config.appName}/users`)
+          .query({
+            $filter: `data/teams/iv/id eq '${teamId}'`,
+          })
+          .reply(200, userResponse);
+
+        const result = await teams.fetchById(teamId, mockUser);
+
+        const expectedResponse = {
+          ...fetchTeamByIdExpectation,
+          members: [
+            {
+              ...fetchTeamByIdExpectation.members[0],
+              avatarUrl: undefined,
+            },
+          ],
+        };
 
-      expect(result).toEqual(expectedResponse);
+        expect(result).toEqual(expectedResponse);
+      });
     });
 
-    test('Should parse the team user correctly when the avatar is undefined', async () => {
+    test('Should return more than a single team on the nested research-output', async () => {
```

Comment:
I know you wanted to make it clear that we have a test case for the new functionality you added, but maybe we should make the assertion more explicit, otherwise it is just a duplicate with: https://github.com/yldio/asap-hub/blob/238050457b37380277be8eb4524699bd66498447/apps/asap-server/test/controllers/teams.test.ts#L251